### DISCUSSION TITLE
Platformio databus class definition trouble

### DISCUSSION THREAD

**Discussion by Deadeye5589** [moononournation/Arduino_GFX#287] [Q&A] [answered]
Hello,

I'm trying to use Platformio under Visual Studio Code to compile a RP2040 project that works with the Arduino IDE. But with Platformio I always get an error that I don't understand 

```
error: expected type-specifier before 'Arduino_RPiPicoSPI' 
Arduino_DataBus *bus = new Arduino_RPiPicoSPI(12 /* DC */, 13 /* CS */, 10 /* SCK */, 11 /* MOSI */, UINT8_MAX /* MISO */, spi1 /* spi */);
```
Any idea what Platformio wants me to do with this type of error?

**Comment by moononournation** [comment]
If the problem only occur in platformIO but no problem in Arduino IDE, then you should ask in platformIO community.

**Reply by Deadeye5589** [reply]
Ok I will try, but maybe you can answer me where the define TARGET_RP2040 is normally to be set from? It does not seem to be part of this libarary. There are if conditions checking this define in the source code but non that is actually setting it.

**Reply by moononournation** [reply]
no idea, i am not using platform io

**Reply by Deadeye5589** [reply, marked as answer]
Well, I have the anwer. It's a bug in the RPi board definition by earlephilhower. The build flag is missing in Platformio. At the moment build_flags = -DTARGET_RP2040 must be added to the platformio.ini to make a project compileable.

**Comment by Deadeye5589** [comment]
Found the fault in the RPi Board definition
